extensions/wikia/PortableInfobox/services/Helpers/PortableInfoboxClassification.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace Wikia\PortableInfobox\Helpers;
+
+class PortableInfoboxClassification {
+
+	/**
+	 * Checks if a page contains non-portable infoboxes based on its title and content.
+	 * The first check is if the title includes a word "infobox":
+	 * - if yes, check for a new markup (<infobox>) in the content. If it is missing - it indicates that
+	 *   the template contains a non-portable infobox
+	 * - if no, check for an occurrence of a word "infobox" inside all class HTML attributes
+	 *
+	 * Returns true if a page may consist a non-portable infobox
+	 *
+	 * @param string $titleText
+	 * @param string $contentText
+	 * @return bool
+	 */
+	public static function isTitleWithNonportableInfobox( $titleText, $contentText ) {
+		// ignore docs pages
+		if ( stripos( $titleText, '/doc' ) ) {
+			return false;
+		}
+
+		$titleNeedle = 'infobox';
+		if ( strripos( $titleText, $titleNeedle ) !== false ) {
+			$portableInfoboxNeedle = '<infobox';
+
+			// If a portable infobox markup was found
+			// it means that the template doesn't have a non-portable infobox
+			return strpos( $contentText, $portableInfoboxNeedle ) === false;
```

Comment:
It's not very important at all, but I would rephrase this comment to:
`if a portable infobox was not found it means that the template has a non-portable one` because this is what we check here.